cos-fleetshard-sync/src/main/java/org/bf2/cos/fleetshard/sync/resources/ConnectorStatusUpdater.java

```diff
@@ -87,11 +89,13 @@ public void update(ManagedConnector connector) {
     }
 
     /*
-     * Expose a metric "cos_fleetshard_sync_connector_state" which reveals the current connector state.
+     * Expose a Gauge metric "cos_fleetshard_sync_connector_state" which reveals the current connector state.
      * Metric value of 1 implies that the connector is in Ready state. Similarly, 2 -> Failed, 3 -> Deleted,
      * 4 -> Stopped, 5 -> In Process
+     * Also exposing a Counter metrics "cos_fleetshard_sync_connector_state_count_total" which reveals each
+     * state count for the connector
      */
-    private void measure(ManagedConnector connector, Integer connectorState) {
+    private void measure(ManagedConnector connector, int connectorState) {
```

Comment:
Maybe also include the `ConnectorDeploymentStatus` previously computed so we can avoid an additional `ConnectorStatusExtractor.extract`